sormas-backend/src/main/java/de/symeda/sormas/backend/environment/EnvironmentService.java

```diff
@@ -249,6 +267,83 @@ public Predicate buildCriteriaFilter(EnvironmentCriteria environmentCriteria, En
 		return filter;
 	}
 
+	private Predicate buildSimilarityFilters(
+		EnvironmentCriteria criteria,
+		CriteriaBuilder cb,
+		Root<Environment> root,
+		EnvironmentQueryContext queryContext) {
+
+		// Environment media and GPS coordinates must always be present for environments to be considered as duplicates
+		if (criteria.getEnvironmentMedia() == null
+			|| criteria.getGpsLatFrom() == null
+			|| criteria.getGpsLatTo() == null
+			|| criteria.getGpsLonFrom() == null
+			|| criteria.getGpsLonTo() == null) {
+			return cb.disjunction();
+		}
+
+		EnvironmentJoins joins = queryContext.getJoins();
+
+		Predicate filter = createDefaultFilter(cb, root);
+
+		// Environment media
+		filter = CriteriaBuilderHelper.and(
+			cb,
+			filter,
+			cb.and(
+				cb.isNotNull(root.get(Environment.ENVIRONMENT_MEDIA)),
+				cb.equal(root.get(Environment.ENVIRONMENT_MEDIA), criteria.getEnvironmentMedia())));
+
+		// GPS coordinates
+		filter = CriteriaBuilderHelper.and(
+			cb,
+			filter,
+			cb.and(
+				cb.isNotNull(joins.getLocation().get(Location.LATITUDE)),
+				cb.greaterThanOrEqualTo(joins.getLocation().get(Location.LATITUDE), criteria.getGpsLatFrom() - 0.02),
+				cb.lessThanOrEqualTo(joins.getLocation().get(Location.LATITUDE), criteria.getGpsLatTo() + 0.02)));
+		filter = CriteriaBuilderHelper.and(
+			cb,
+			filter,
+			cb.and(
+				cb.isNotNull(joins.getLocation().get(Location.LONGITUDE)),
+				cb.greaterThanOrEqualTo(joins.getLocation().get(Location.LONGITUDE), criteria.getGpsLonFrom() - 0.02),
+				cb.lessThanOrEqualTo(joins.getLocation().get(Location.LONGITUDE), criteria.getGpsLonTo() + 0.02)));
```

Comment:
GPS latitude and longitude can take also negative values in this case the evaluation of "-0.02" and "+0.02" should work the other way around. Also, there is a situation when you are within less than 0.02 from the equator or from the prime meridian.